evap/staff/views.py

```diff
@@ -626,40 +652,65 @@ def course_participant_import(request, semester_id, course_id):
     course = get_object_or_404(Course, id=course_id, semester=semester)
     raise_permission_denied_if_archived(course)
 
-    form = UserImportForm(request.POST or None, request.FILES or None)
+    excel_form = UserImportForm(request.POST or None, request.FILES or None)
+    copy_form = CourseParticipantCopyForm(request.POST or None, request.FILES or None)
 
-    if form.is_valid():
+    if request.method == "POST":
         operation = request.POST.get('operation')
-        if operation not in ('test', 'import'):
-            raise SuspiciousOperation("Invalid POST operation")
 
-        # Extract data from form.
-        excel_file = form.cleaned_data['excel_file']
-        import_course = form.cleaned_data['course']
+        if operation not in ('test', 'import', 'copy'):
+            raise SuspiciousOperation("Invalid POST operation")
 
         test_run = operation == 'test'
+        import_run = operation == 'import'
+        copy_run = operation == 'copy'
+
+        if test_run and excel_form.is_valid():
+            excel_file = excel_form.cleaned_data['excel_file']
+            imported_users, warnings, errors = UserImporter.process(request, excel_file, test_run)
+            if not errors:
+                save_import_file(excel_file, str(request.user.id))
+                # messages.success(request, "%d Participants would be added to course %s" % (len(imported_users), course.name))
```

Comment:
The commented code was a result of the UserImporter.process method currently not returning a correct list if you're doing a test run, thus always resulting in the message "0 Users would be imported." I changed the comment from the code to an explanation and I will add a reference on Issue #724